utils/appliance/__init__.py

```diff
@@ -249,37 +250,103 @@ def seal_for_templatizing(self):
             ssh_client.run_command('echo "vm.swappiness = 1" >> /etc/sysctl.conf',
                 ensure_host=True)
 
+    def _encrypt_string(self, string):
+        try:
+            # Let's not log passwords
+            logging.disable(logging.CRITICAL)
+            rc, out = self.ssh_client.run_rails_command(
+                "puts MiqPassword.encrypt('{}')".format(string))
+            return out
+        finally:
+            logging.disable(logging.NOTSET)
+
+    def _get_ems_ips(self, ems):
```

Comment:
i would have liked to suggest the ipaddress module, but its gleefully broken on python2 insisting on unicode strings
we could however go as far as using it when the database returns that to us